airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/generate_erd/pipeline.py

```diff
@@ -0,0 +1,266 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+from __future__ import annotations
+
+import json
+import os
+from pathlib import Path
+from typing import TYPE_CHECKING, List, Optional
+
+import dpath
+import google.generativeai as genai
+
+from airbyte_protocol.models import AirbyteCatalog
+from dagger import Container
+from markdown_it import MarkdownIt
+from pipelines.airbyte_ci.connectors.build_image.steps.python_connectors import BuildConnectorImages
+from pipelines.airbyte_ci.connectors.consts import CONNECTOR_TEST_STEP_ID
+from pipelines.airbyte_ci.connectors.context import ConnectorContext, PipelineContext
+from pipelines.airbyte_ci.connectors.generate_erd.dbml_assembler import Source, DbmlAssembler
+from pipelines.airbyte_ci.connectors.generate_erd.relationships import RelationshipsMerger, Relationships
+from pipelines.airbyte_ci.connectors.reports import Report
+from pipelines.consts import LOCAL_BUILD_PLATFORM
+from pipelines.helpers.connectors.command import run_connector_steps
+from pipelines.helpers.execution.run_steps import STEP_TREE, StepToRun
+from pipelines.models.steps import Step, StepResult, StepStatus
+from pydbml.renderer.dbml.default import DefaultDBMLRenderer
+
+if TYPE_CHECKING:
+    from anyio import Semaphore
+
+# TODO: pass secret in dagger?
```

Comment:
I will sync tomorrow with @alafanechere for what is the best way to manage those secrets. In the meanwhile, the users will have to copy/paste them here

Reply:
The "best way" is to add a `--gen-ai-api-key` option to your `generate_erd` command: 

```
@click.option(
    "--gen-ai-api-key,
    help="Your GEN AI API key",
    type=click.STRING,
    required=True,
    envvar="GEN_AI_API_KEY",
    callback=wrap_in_secret,
)
```

This will add `gen_api_key` to our in memory secret store.
You can access it via:
```
Secret("gen_ai_api_key", ctx.obj["secret_stores"]["in_memory"])
```

You would have to add an optional `gen_ai_api_key` attribute to the `ConnectorContext` class and pass it from the click context to this class in [`get_connector_context`](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/helpers/connectors/command.py#L16)

and finally in your `GenerateErdSchema` step the secret value should be reachable like this:
`self.context.gen_ai_api_key.value`

Reply:
That makes sense to me. At that point if we want to make it part of the CI, it would only mean that we have to pass them through the GitHub secrets, right?